### PR TITLE
Add new bulk create constructor without optional _id field

### PR DESCRIPTION
**Pull request by camerondurham**
### Change (latest - revision 2)

Added new `create_no_id` and `new_no_id` constructors/helpers to support bulk create operations where no _id field is required.

#### Change (revision 1)

Changed types to `id: Option<S>` when `_id` is optional in the bulk request, otherwise, keep `id` parameter as `String`.

This change is making the Bulk operation constructor parameters more closely match the Bulk API, specifically when `_id` is required or not.

Summarized from the docs: https://www.elastic.co/guide/en/elasticsearch/reference/7.10/docs-bulk.html#bulk-api-request-body
- `create`: `_id` (optional, string)
- `delete`: `_id` (required, string)
- `index`: `_id` (optional, string)
- `update` `_id` (required, string)

### Issues Resolved

Related to, and may close issue #241 .

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
@Xtansia take a look?

Regarding a breaking change: if it can be avoided it's best. Then we don't need to do a major release and break users. You can deprecate the old method.

**Comment by Xtansia** [comment]
I also don't feel this warrants a breaking version bump just for this, however there is an opportunity where you could PR a non-breaking change directly into `2.x` and make the breaking change in `main`. Then it'll be usable sooner via a minor version, and the "corrected" version can just wait for whenever we cut a 3.0.0

File: opensearch/src/root/bulk.rs [hunk 1]
```diff
@@ -179,6 +179,11 @@ where
         BulkCreateOperation::new(id, source)
     }
 
+    /// Creates a new instance of [bulk create operation](BulkCreateOperation) without explicit id
+    pub fn create_no_id(source: B) -> BulkCreateOperation<B> {
+        BulkCreateOperation::new_no_id(source)
```
Comment (Xtansia) [inline opensearch/src/root/bulk.rs:184]: I'd prefer `without_id` over `no_id` for comprehensibility.
```suggestion
    /// Creates a new instance of [bulk create operation](BulkCreateOperation) without an explicit id
    pub fn create_without_id(source: B) -> BulkCreateOperation<B> {
        BulkCreateOperation::new_without_id(source)
```

File: opensearch/src/root/bulk.rs [hunk 2]
```diff
@@ -245,6 +250,22 @@ impl<B> BulkCreateOperation<B> {
         }
     }
 
+    /// Creates a new instance of [BulkCreateOperation] without explicit id
+    pub fn new_no_id(source: B) -> Self {
```
Comment (Xtansia) [inline opensearch/src/root/bulk.rs:254]: I'd prefer `without_id` over `no_id` for comprehensibility.
```suggestion
    /// Creates a new instance of [BulkCreateOperation] without an explicit id
    pub fn new_without_id(source: B) -> Self {
```

File: opensearch/src/root/bulk.rs [hunk 3]
```diff
@@ -789,6 +810,7 @@ mod tests {
                 .routing("routing"),
         )?;
         ops.push(BulkOperation::create("2", CreateDoc { bar: "create" }))?;
+        ops.push(BulkOperation::create_no_id(CreateDoc { bar: "create" }))?;
```
Comment (Xtansia) [inline opensearch/src/root/bulk.rs:813]:
```suggestion
        ops.push(BulkOperation::create_without_id(CreateDoc { bar: "create" }))?;
```

**Review by Xtansia — CHANGES_REQUESTED** [review]
Changes look good, I just have some thoughts about the naming. 
Could you please also add an entry to the [CHANGELOG.md under the "Added" header](https://github.com/opensearch-project/opensearch-rs/blob/main/CHANGELOG.md#added).
You can ignore the `Link Checker` workflow failure.

**Comment by camerondurham** [comment]
Thanks for the review, I accepted all suggestions and added an entry to the changelog.

**Review by Xtansia — APPROVED** [review]

**Review by reta — APPROVED** [review]

**Pull request merged**